interfaces/builtin/snapd_observe.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
+)
+
+const snapdObserveSummary = `allows observing the snapd API`
+
+const snapdObserveBaseDeclarationPlugs = `
+  snapd-observe:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const snapdObserveBaseDeclarationSlots = `
+  snapd-observe:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+var supportedApis = []string{"/v2/changes"}
+
+func (iface *snapdObserveInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	api, ok := plug.Attrs["api"].(string)
```

Comment:
this would ideally be a list no?

Reply:
Yes, but I removed it now as suggested in the top-level comments